azure-devops/create-new-agent-image.ps1

```diff
@@ -0,0 +1,297 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# This script assumes you have installed Azure tools into Powershell by following the instructions
+# at https://docs.microsoft.com/en-us/powershell/azure/install-az-ps?view=azps-3.6.1
+# or are running from Azure Cloud Shell
+#
+
+$Location = 'westus2'
+$Prefix = 'StlBuild-' + (Get-Date -Format 'yyyy-MM-dd')
+$VMSize = 'Standard_D16s_v3'
+$ProtoVMName = 'PROTOTYPE'
+$LiveVMPrefix = 'BUILD'
+$WindowsServerSku = '2019-Datacenter'
+
+$ProgressActivity = 'Creating Scale Set'
+$TotalProgress = 8
+$CurrentProgress = 1
+
+function Find-ResourceGroupName {
+  Param(
+    [string] $prefix
+  )
+
+  $result = $prefix
+  $resources = Get-AzResourceGroup
+  $collision = $false
+  foreach ($resource in $resources) {
+    if ($resource.ResourceGroupName -eq $result) {
+      $collision = $true
+      break
+    }
+  }
+
+  $suffix = 0
+  while ($collision) {
+    $collision = $false
+    $suffix++
+    $result = "$prefix-$suffix"
+    foreach ($resource in $resources) {
+      if ($resource.ResourceGroupName -eq $result) {
+        $collision = $true
+        break
+      }
+    }
+  }
+
+  return $result
+}
+
+function New-Password {
+  Param (
+    [int] $length = 32
+  )
+
+  $Chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
+
+  $result = ''
+  for ($idx = 0; $idx -lt $length; $idx++) {
+    $result += $Chars[(Get-Random -Minimum 0 -Maximum ($Chars.Length - 1))]

```

Comment:
I don't think this can ever generate `'9'`. According to https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/get-random?view=powershell-7 , "`Get-Random` returns a value that is less than the maximum (not equal)." Can you verify?

Reply:
```
PS D:\STL> for ($idx = 0; $idx -lt 10000; $idx++) { if ((Get-Random -Minimum 0 -Maximum 1) -eq 1) { 
>> Write-Output Pass
>> return
>> }}
PS D:\STL>
```

Talk about terrible parameter naming.